### PR TITLE
started testing on the backend

### PR DESCRIPTION
**Pull request by Albertamoahjunior**
I have began testing on the backend
- I started with the auth routes/endpoints

### CODE REVIEW

File: backend/__tests__/routes.test.ts
Comment (genes2160) [inline backend/__tests__/routes.test.ts:1]: In your opinion, does all the post request ends up persisting data to "the main DB" ?

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: Please I do not get the question well

Comment (genes2160) [reply backend/__tests__/routes.test.ts:1]: which part dont you get?

Comment (genes2160) [reply backend/__tests__/routes.test.ts:1]:
```
  //describe register user
  describe('POST /register', () =>{
    it('should register new users', async () =>{
      const res = await request(app)
      .post('/register')
      .send({username:'Kwadwo', password: 'asamoah24'});

      expect(res.status).toBe(200);
      expect(res.body).toHaveProperty('data');
    });
  });
```

Does this persist this user to the DB in your opinion ?

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: Yes please

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: Wait, please there is a caveat

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: After you run the tests for the first time, you would have to change the details others you will get a 400 error: user already exists

Comment (genes2160) [reply backend/__tests__/routes.test.ts:1]: Remember one of the objective is to not persist that during tests to the actual DB

Comment (genes2160) [reply backend/__tests__/routes.test.ts:1]: The 400 can be a use case to create random users if exists this should not fail the test but acknowledge use case can be flagged

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: Yes please

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: That means it has become imperative to spin on the fly database so I can delete after the tests

Comment (genes2160) [reply backend/__tests__/routes.test.ts:1]: Just look into => beforeEach and afterEach hooks Jest

implement this layer on what has been done - its pretty simple

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: but please it is the db service that has access to the database, and I do not know how to go about that without using the db service. Or I can mock the db service functions.

Comment (Albertamoahjunior) [reply backend/__tests__/routes.test.ts:1]: ahhnn I have seen that. 
Thank you

**Review by genes2160 — APPROVED** [review]

**Pull request merged**